packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx

```diff
@@ -765,6 +765,56 @@ function MapAnalysisImpl(props: ImplProps) {
     'single-variable-' + appState.activeMarkerConfigurationType
   );
 
+  // snackbar
+  const { enqueueSnackbar } = useSnackbar();
+
+  // setting button behavior at Legend
+  const openPanelBehavior = () => {
+    // for barplot, activeSideManuId is based on 'bar', but activeMarkerConfigurationType is 'barplot'
+    setActiveSideMenuId(
+      'single-variable-' +
+        (appState.activeMarkerConfigurationType === 'barplot'
+          ? 'bar'
+          : appState.activeMarkerConfigurationType)
+    );
+    setIsSidePanelExpanded(true);
+  };
+
+  const openPanel = useCallback(() => {
+    appState.isSidePanelExpanded
+      ? enqueueSnackbar('Marker configuration panel is already open', {
+          variant: 'info',
+        })
+      : openPanelBehavior();
+  }, [
```

Comment:
There are a bunch of "indirect" dependencies listed here.

I would probably be better to either create `openPanelBehavior` with useCallBack with the dependency [`appState.activeMarkerConfigurationType`] (the useState-derived setters are stable and not needed), and then add `openPanelBehavior` as a dependency for the `openPanel` useCallback

Or... Just paste the code from `openPanelBehavior` into `openPanel`, double-checking the necessity of the dependencies as mentioned above. This would be fine because I don't see `openPanelBehavior` used elsewhere.

Reply:
Hi @bobular! Thank you for your detailed reviews 👍 Actually I noticed that indirect dependencies this morning and as you suggested I think we can just use single openPanel useCallback. Updated accordingly.

Also, I just addressed the first item, "There's just one behavioural problem, where if the side panel is already expanded and on, say, the Filter panel, then it won't switch to the marker config panel when the gear icon is clicked." It is now switched to marker configuration with snackbar behavior. I think that the other UI/UXs tuffs can be handled in other ticket #785 as you suggested.

Reply:
Hi - that's still not quite the correct behaviour, IMO.

I was thinking more like this

```
  const openPanel = useCallback(() => {
    const nextSideMenuId = 'single-variable-' +
           (appState.activeMarkerConfigurationType === 'barplot'
          ? 'bar'
          : appState.activeMarkerConfigurationType);
    
    if (appState.isSidePanelExpanded && activeSideMenuId === nextSideMenuId)
      enqueueSnackbar('Marker configuration panel is already open', {
        variant: 'info',
      });
    if (!appState.isSidePanelExpanded)
      setIsSidePanelExpanded(true);
    if (activeSideMenuId !== nextSideMenuId)
      setActiveSideMenuId(nextSideMenuId);
  }, [
    appState.isSidePanelExpanded,
    enqueueSnackbar,
    activeSideMenuId,
    setIsSidePanelExpanded,
    appState.activeMarkerConfigurationType,
  ]);
```